src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+				'body'     => $data,
+				'compress' => true,
+			]
+		);
+
+		if ( ! is_wp_error( $response ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $response ), true );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * To get amp data to send it to compatibility server.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array
+	 */
+	public function get_data() {
+
+		$amp_urls = $this->get_amp_urls();
+
+		$request_data = [
+			'site_url'      => static::get_home_url(),
+			'site_info'     => $this->get_site_info(),
+			'plugins'       => $this->get_plugin_info(),
+			'themes'        => $this->get_theme_info(),
+			'errors'        => array_values( $amp_urls['errors'] ),
+			'error_sources' => array_values( $amp_urls['error_sources'] ),
+			'urls'          => array_values( $amp_urls['urls'] ),
+			'error_log'     => $this->get_error_log(),
+		];
+
+		if ( ! empty( $this->args['is_synthetic'] ) ) {
+			$request_data['site_info']['is_synthetic_data'] = true;
+		}
+
+		$this->data = $request_data;
+
+		return $request_data;
+	}
+
+	/**
+	 * Normalize a URL for storage.
+	 *
+	 * The AMP query param is removed to facilitate switching between standard and transitional.
+	 * The URL scheme is also normalized to HTTPS to help with transition from HTTP to HTTPS.
+	 *
+	 * @since 2.2
+	 *
+	 * @reference AMP_Validated_URL_Post_Type::normalize_url_for_storage
+	 *
+	 * @param string $url URL.
+	 *
+	 * @return string Normalized URL.
+	 */
+	public static function normalize_url_for_storage( $url ) {
+
+		// Only ever store the canonical version.
+		if ( ! amp_is_canonical() ) {
+			$url = amp_remove_paired_endpoint( $url );
+		}
+
+		// Remove fragment identifier in the rare case it could be provided. It is irrelevant for validation.
+		$url = strtok( $url, '#' );
+
+		// Query args to be removed from validated URLs.
+		$removable_query_vars = array_merge(
+			wp_removable_query_args(),
+			[ 'preview_id', 'preview_nonce', 'preview', QueryVar::NOAMP ]
+		);
+
+		// Normalize query args, removing all that are not recognized or which are removable.
+		$url_parts = explode( '?', $url, 2 );
+		if ( 2 === count( $url_parts ) ) {
+			$args = wp_parse_args( $url_parts[1] );
+			foreach ( $removable_query_vars as $removable_query_arg ) {
+				unset( $args[ $removable_query_arg ] );
+			}
+			$url = $url_parts[0];
+			if ( ! empty( $args ) ) {
+				$url = $url_parts[0] . '?' . build_query( $args );
+			}
+		}
+
+		// Normalize the scheme as HTTPS.
+		$url = set_url_scheme( $url, 'https' );
+
+		return $url;
+	}
+
+	/**
+	 * To get site info.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array Site information.
+	 */
+	public function get_site_info() {
+
+		global $wpdb;
+
+		$wp_type = is_multisite() ? ( defined( 'SUBDOMAIN_INSTALL' ) && SUBDOMAIN_INSTALL ) ? 'subdomain' : 'subdir' : 'single';
+
+		$active_theme = wp_get_theme();
+		$active_theme = static::normalize_theme_info( $active_theme );
+
+		$amp_settings = \AMP_Options_Manager::get_options();
```

Comment:
Better to add `use AMP_Options_Manager;` up top and then:

```suggestion
		$amp_settings = AMP_Options_Manager::get_options();
```

Reply:
Same goes for the other global-namespaced classes elsewhere in this file.